src/apps/common/submodules/chooseModel/chooseModel.js

```diff
@@ -35,40 +47,39 @@ define(function(require) {
 		},
 
 		chooseModelFormatProvisionerData: function(data) {
-			var formattedData = {
-					brands: []
-				},
-				families,
-				models;
-
-			_.each(data, function(brand, brandKey) {
-				families = [];
-
-				_.each(brand.families, function(family, familyKey) {
-					models = [];
-
-					_.each(family.models, function(model, modelKey) {
-						models.push({
-							id: modelKey.toLowerCase(),
-							name: model.name
-						});
-					});
-
-					families.push({
-						id: familyKey.toLowerCase(),
-						name: family.name,
-						models: models
-					});
-				});
-
-				formattedData.brands.push({
-					id: brandKey.toLowerCase(),
-					name: brand.name,
-					families: families
-				});
-			});
-
-			return formattedData;
+			var self = this,
+				resolveModelName = function resolveModelName(familyName, modelName) {
+					return _
+						.chain(self.appFlags.chooseModel.familyNamesAsPrefix)
+						.filter(function(name) {
+							return name === familyName;
+						})
+						.map(_.toUpper)
+						.concat([modelName])
+						.join(' ')
+						.value();
+				};
+
+			return {
+				brands: _.map(data, function(brand, brandKey) {
+					return {
+						id: _.lowerCase(brandKey),
+						name: brand.name,
+						families: _.map(brand.families, function(family, familyKey) {
+							return {
+								id: _.toLower(familyKey),
+								name: family.name,
+								models: _.map(family.models, function(model, modelKey) {
+									return {
+										id: _.toLower(modelKey),
+										name: resolveModelName(family.name, model.name)
```

Comment:
Thanks for that, Looks great to me.

I'm curious about something.

How this should works for prefixes that are not included in `familyNamesAsPrefix` array?
Looking at `resolveModelName` this method looks for prefixes to return the model name, but if the model family isn't in the list, what this method will return, since this is filtering by `familyName` I assume an empty value `string`?

Please, let me know if I'm missing something :)

@jrmende5 @azefiel

Reply:
`lodash#filter` returns an array not matter what, even when no item pass the test.

So when the model's family is not included in `familyNamesAsPrefix`, then an empty array will be concatenated to the array containing the model name, which is joined and gives a final string representing the model's name to display to the end-user.

Reply:
Ah, that makes more sense now, thanks for clarify.